src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -168,8 +205,11 @@ private static boolean isValidSpin(final int numOfSpins) {
     private static CharSequence doPeek(final CharSequence pattern) {
         CharSequence returnPattern;
         boolean validPattern = isPeekValid(pattern);
-
-        if(validPattern) {
+        if (dev == null) {
```

Comment:
You can move this to isPeekValid() and then change this to 

`if(isPeekValid() && validPattern) {
    //doPeek
} else  {
//append and set return pattern
}
return returnPattern
`

Reply:
I think you mean to have `isPeekValid` to ensure device is not null?